### ISSUE TITLE
sys_usbd: Disney Infinity Base (VID:0E6F/PID:0129)

### ISSUE THREAD

**Issue by AniLeo** [RPCS3/rpcs3#8010]
```
USB\VID_0E6F&PID_0129&REV_0200
USB\VID_0E6F&PID_0129
```

Requires whitelisting and testing

**Comment by Demolitan** [comment]
When booting up the game, RPCS3 recognises the Disney Infinity Portal, but as soon as the game starts it freezes indefinitely.
Changing the GPU/CPU settings has no effect.
Multiple people seem to have this issue.

**Comment by illusion0001** [comment]
Missing log file

**Comment by Demolitan** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/4470268/RPCS3.log)

**Comment by AniLeo** [comment]
```
·W 0:00:27.173129 {PPU[0x1000000] Thread (main_thread) [0x022d4d60]} sys_ppu_thread: _sys_ppu_thread_create(): Thread “libusbd_usbd_thread” created (id=0x1000015, func=*0x28123cc, rtoc=0x2828340, user-tls=0x2000b0b0)
·W 0:00:27.173261 {PPU[0x1000000] Thread (main_thread) [0x02815454]} sys_usbd: sys_usbd_get_device_list(handle=0x115b, device_list=*0x374f65d0, max_devices=0x7f)
·W 0:00:27.173296 {PPU[0x1000000] Thread (main_thread) [0x02816e2c]} sys_usbd: sys_usbd_register_extra_ldd(handle=0x115b, s_product=“Gateway”, slen_product=0x7, id_vendor=0xe6f, id_product_min=0x129, id_product_max=0x129)
·S 0:00:27.173331 {PPU[0x1000000] Thread (main_thread) [0x02816e2c]} sys_usbd: Ldd device matchup for <Gateway>
...
·U 0:00:27.178946 {PPU[0x1000015] Thread (libusbd_usbd_thread) [0x02812b44]} sys_usbd TODO: sys_usbd_attach(handle=0x115b)
```

Unsure if it's a failure caused by sys_usbd since the device is detected

**Comment by Demolitan** [comment]
Could the fact that the Skyldanders portal (the portal the code was written for initially) has 1 nfc port whereas the Disney Portal has 3, play a part here?

**Comment by Demolitan** [comment]
Update: After updating to 10201, the game runs and recognises the base the first time you boot it up. However every other attempt after that, the game freezes.

**Comment by Coloradohusky** [comment]
Both Disney 2.0 and 3.0 freeze as well
[2.0.log](https://github.com/RPCS3/rpcs3/files/4475622/2.0.log)
[3.0.log](https://github.com/RPCS3/rpcs3/files/4475623/3.0.log)

**Comment by helpmeree** [comment]
bruh i wanna whitelist it just tell me how to do it

**Comment by MSuih** [comment]
It has been whitelisted on emulator side, so just update to latest version. On windows you have to use Zadig to get WinUSB drivers for it.

**Comment by Demolitan** [comment]
Yes. The emulator recognises the device when you change the driver with zadig, but when you boot the game it immediately crashes, so I don’t think it is the whitelisting. Probably a software issue more likely.

**Comment by mainzerbe** [comment]
The latest RPCS3 release works fine with Disney Infinity and Disney Infinity 2.0 (incl. portal functionality besides graphical issues) but Disney Infinity 3.0 runs only up to release 0.0.12-10997. Unfortunately the bug freezing the game after start when the portal is connected isn't solved in 0.0.12-10997 (U {PPU[0x1000013] Thread (libusbd_usbd_thread) [0x03a12b44]} sys_usbd TODO: sys_usbd_attach(handle=0x115b)). Later builds crash in the building of the PPU modules.

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9924286/RPCS3.log)
